internal/datastore/crdb/crdb.go

```diff
@@ -159,11 +154,101 @@ type crdbDatastore struct {
 	*revisions.RemoteClockRevisions
 
 	dburl             string
-	conn              *pgxpool.Pool
+	pool              *pgxpool.Pool
 	watchBufferLength uint16
-	querySplitter     common.TupleQuerySplitter
+	writeOverlapKeyer overlapKeyer
+	usersetBatchSize  uint16
 	execute           executeTxRetryFunc
-	overlapKeyer      overlapKeyer
+}
+
+func (cds *crdbDatastore) SnapshotReader(rev datastore.Revision) datastore.Reader {
+	createTxFunc := func(ctx context.Context) (pgx.Tx, common.TxCleanupFunc, error) {
+		tx, err := cds.pool.BeginTx(ctx, pgx.TxOptions{AccessMode: pgx.ReadOnly})
+		if err != nil {
+			return nil, nil, err
+		}
+
+		cleanup := func(ctx context.Context) {
+			if err := tx.Rollback(ctx); err != nil {
+				log.Ctx(ctx).Err(err).Msg("error rolling back transaction")
```

Comment:
Add additional context to the error message, since it is the same message as below, and once line numbers move, it'll be harder to pin down the exact location if we see the error occur

Reply:
i mean they will have different underlying errors.